tox.ini

```diff
@@ -1,16 +1,28 @@
 [tox]
-envlist = py37,pypy3,py38,py39,py310,py311,p312
+envlist = py37,pypy3,py38,py39,py310,py311,p312,sqla2,sqla13
 
 [testenv]
 groups = dev,test
 commands =
     pytest tests {posargs}
 
+[testenv:sqla2]
+groups = dev,test
+commands =
+    python -I -m pip install 'sqlalchemy>=2.0.0'
+    pytest tests {posargs}
+
+[testenv:sqla13]
+groups = dev,test
+commands =
+    python -I -m pip install 'sqlalchemy>=1.3,<1.4'
```

Comment:
so 1.4 in particular is not tested ? is this intentional ?

Reply:
My intention was to test all of python versions with 1.4 (as it is installed in the default testenv), test 1.3 againts some older versions and to test 2 againts newer versions of python

Reply:
for instance, 3.11 pipeline has two runs of tox:
https://github.com/evo-company/hiku/actions/runs/10851816805/job/30117285246?pr=163 

```
py311: aiopg==1.4.0,astor==0.8.1,async-timeout==4.0.3,asyncpg==0.28.0,black==23.3.0,certifi==2023.7.22,click==8.1.7,coverage==7.2.7,Faker==18.13.0,flake8==5.0.4,graphql-core==3.2.3,greenlet==3.0.0,grpcio==1.59.0,grpcio-tools==1.59.0,hiku @ file:///home/runner/work/hiku/hiku/.tox/.tmp/package/1/hiku-0.7.4.tar.gz#sha256=a344ff2d0fe4054ddb5f7bcf0276030ef8f2e279ab6ea29e8a5f8dd3e7f61d8d,iniconfig==2.0.0,mccabe==0.7.0,mypy==1.4.1,mypy-extensions==1.0.0,packaging==23.1,pathspec==0.11.2,pip==24.2,platformdirs==3.10.0,pluggy==1.2.0,prometheus-client==0.17.1,protobuf==4.24.4,psycopg2-binary==2.9.7,py-cpuinfo==9.0.0,pycodestyle==2.9.1,pyflakes==2.5.0,pygal==3.0.0,pygaljs==1.0.2,pytest==7.4.2,pytest-asyncio==0.21.1,pytest-benchmark==4.0.0,pytest-cov==4.1.0,python-dateutil==2.8.2,sentry-sdk==1.31.0,setuptools==68.0.0,six==1.16.0,SQLAlchemy==1.4.49,typing_extensions==4.7.1,urllib3==2.0.4,wheel==0.44.0
tox: py311
py311: OK ✔ in 30.35 seconds
```

```
.pkg: _exit> python /opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/pyproject_api/_backend.py True pdm.backend
  py311: OK (30.35=setup[27.07]+cmd[3.28] seconds)
  sqla2: OK (30.00=setup[25.56]+cmd[1.93,2.52] seconds)
```

Reply:
I used pip install as one of the commands instead of specifying sqlalchemy via tox `deps` because these deps are installed before running pdm which overwrites them